tools/make.py

```diff
@@ -173,6 +186,11 @@
                       default=False,
                       help="Link with mbed test library")
 
+    parser.add_argument("--metadata",
```

Comment:
how long does it take to generate this? costs nothing? The build could contain this. Or do we consider this to be part of logging (=verbose) ?